utils/env.sh

```diff
@@ -122,18 +126,28 @@ function appendEnvPartialIfExists () {
         "${WARDEN_DIR}/environments/includes/${PARTIAL_NAME}.base.yml" \
         "${WARDEN_DIR}/environments/includes/${PARTIAL_NAME}.${WARDEN_ENV_SUBT}.yml" \
         "${WARDEN_DIR}/environments/${WARDEN_ENV_TYPE}/${PARTIAL_NAME}.base.yml" \
-        "${WARDEN_DIR}/environments/${WARDEN_ENV_TYPE}/${PARTIAL_NAME}.${WARDEN_ENV_SUBT}.yml" \
         "${WARDEN_HOME_DIR}/environments/includes/${PARTIAL_NAME}.base.yml" \
         "${WARDEN_HOME_DIR}/environments/includes/${PARTIAL_NAME}.${WARDEN_ENV_SUBT}.yml" \
         "${WARDEN_HOME_DIR}/environments/${WARDEN_ENV_TYPE}/${PARTIAL_NAME}.base.yml" \
-        "${WARDEN_HOME_DIR}/environments/${WARDEN_ENV_TYPE}/${PARTIAL_NAME}.${WARDEN_ENV_SUBT}.yml" \
         "${WARDEN_ENV_PATH}/.warden/environments/includes/${PARTIAL_NAME}.base.yml" \
         "${WARDEN_ENV_PATH}/.warden/environments/includes/${PARTIAL_NAME}.${WARDEN_ENV_SUBT}.yml" \
-        "${WARDEN_ENV_PATH}/.warden/environments/${WARDEN_ENV_TYPE}/${PARTIAL_NAME}.base.yml" \
-        "${WARDEN_ENV_PATH}/.warden/environments/${WARDEN_ENV_TYPE}/${PARTIAL_NAME}.${WARDEN_ENV_SUBT}.yml"
+        "${WARDEN_ENV_PATH}/.warden/environments/${WARDEN_ENV_TYPE}/${PARTIAL_NAME}.base.yml"
     do
         if [[ -f "${PARTIAL_PATH}" ]]; then
             DOCKER_COMPOSE_ARGS+=("-f" "${PARTIAL_PATH}")
         fi
     done
+
```

Comment:
@joseluisi4 This doesn't seem like it would work to me. The value of `PARTIAL_NAME` will be "mutagen", but there's no check against the partial name.

I think a better solution would be to check the `PARTIAL_NAME` prior to the for loop, and if it's "mutagen" and mutagen is disabled, then just return from the function; otherwise, process the for-loop as originally developed.

Reply:
`PARTIAL_NAME` will never be "mutagen", it is the name of the **environment**

The problem I am trying to solve is that the `xxx.darwin.yml` files are not used because these volumes are designed for Mutange

Another solution would be to rename those files to `xxx.mutagen.yml`, this way you could filter by name

Reply:
Partial name is _**not**_ the name of the environment. The `appendEnvPartialIfExists` function takes the name of the service partial to include (e.g. `nginx`, or `php`), not the environment. See https://github.com/wardenenv/warden/blob/39d472934585b7a656033f52c5ae4150470912bd/commands/env.cmd#L80

My reservation about removing the OS-specific partials is that if you have an os-specific partial for a service in your environment but aren't using Mutagen, it will never be included.

To me this would be an opportunity to properly refactor the mutagen inclusion to use the `appendEnvPartialIfExists` function to add Mutagen syncs, and only including them if the OS is darwin and Mutagen is enabled. This would then also mean that the checks for starting / pausing Mutagen would need to be adjusted to use the `WARDEN_MUTAGEN_ENABLE` variable.